node/src/actors/peers_manager/handlers.rs

```diff
@@ -54,19 +54,15 @@ impl Handler<RemovePeers> for PeersManager {
 }
 
 /// Handler for GetRandomPeer message
-impl Handler<GetRandomPeer> for PeersManager {
-    type Result = PeersSocketAddrResult;
+impl Handler<GetRandomPeers> for PeersManager {
+    type Result = PeersSocketAddrsResult;
 
-    fn handle(&mut self, _msg: GetRandomPeer, _: &mut Context<Self>) -> Self::Result {
-        let result = self.peers.get_random();
+    fn handle(&mut self, msg: GetRandomPeers, _: &mut Context<Self>) -> Self::Result {
+        let result = self.peers.get_random_peers(msg.n);
 
         match result {
-            Ok(Some(address)) => {
-                log::debug!("Selected a random peer address: {:?}", address);
-                result
-            }
-            Ok(None) => {
-                log::warn!("Could not select a random peer address because there were none");
+            Ok(_) => {
+                log::debug!("Selected random peer addresses");
```

Comment:
Why not list selected peers?:

```suggestion
                log::debug!("Selected random peer addresses: {:?}", peers);
```